src/components/Select/Select.js

```diff
@@ -1,30 +1,69 @@
-import React from 'react';
+import React, { forwardRef } from 'react';
 import PropTypes from 'prop-types';
 
 import { Select as AscSelect } from '@amsterdam/asc-ui';
 
-const Select = ({ label, onChange, name, options, value }) => (
-  <AscSelect value={value} onChange={onChange} data-testid={name} label={label} name={name}>
-    {options.map(option => (
-      <option key={`${name}-${option.key}`} value={option.value}>
-        {option.name}
-      </option>
-    ))}
-  </AscSelect>
+const SelectOptions = ({ name, options, optionKey, optionValue }) =>
+  options.map(option => (
+    <option key={`${name}-${option[optionKey]}`} value={option[optionKey]}>
+      {option[optionValue]}
+    </option>
+  ));
+
+const Select = forwardRef(
+  ({ label, onChange, name, value, options, optionKey = 'key', optionValue = 'name', groups, emptyOption }, ref) => (
+    <AscSelect value={value} onChange={onChange} data-testid={name} label={label} name={name} ref={ref}>
+      {emptyOption && (
+        <option key={`${name}-${emptyOption[optionKey]}`} value={emptyOption[optionKey]}>
+          {emptyOption[optionValue]}
+        </option>
+      )}
+
+      {groups?.length > 1 ?
+        groups?.map(group => (
+          <optgroup key={group.name} label={group.name}>
+            <SelectOptions
+              name={name}
+              options={options.filter(option => option.group === group.value)}
+              optionKey={optionKey}
+              optionValue={optionValue}
+            />
+          </optgroup>
+        ))
+        : (
+          <SelectOptions name={name} options={options} optionKey={optionKey} optionValue={optionValue} />
+        )}
+    </AscSelect>
+  )
 );
 
+Select.defaultProps = {
+  groups: null,
+  emptyOption: null,
```

Comment:
Setting default values to `null` isn't really necessary I imagine.

Reply:
removed!

Reply:
The same goes for `groups` of course.